docs/markdown/DevGuide_BuildProcessor.md

```diff
@@ -0,0 +1,53 @@
+If you have created your new processor using Inviwo-meta, you can see there is already a class definition in your created files.
+Your processor needs to be a subclass of `Processor` and it needs to override the `void Processor::process()` method. The `process()` method is where a processors action is implemented.
+Additionally, you can override the `const ProcessorInfo getProcessorInfo() const` method which returns meta information about the processor, like a class identifier, name, category, tags etc.
+Further your processor needs to define a constructor to initialize all your properties and a destructor (which will usually be the default destructor).
+
+You can then add properties as private attributes in your class. Make sure that you initialize each property in the constructor. We recommend using initializer lists for that.
+In the constructor body itself you have to call `addProperty()` or `addProperties()` to add the properties to your processor. This will make sure the properties are displayed in the specified order.
+
+## Inports and Outports
+Similar to properties, you can add ports as private members of your class and initialize them in the constructor. Ports also need to be added to the processor using `addPort()`. Note that the convention is to put optional inports to the right of the processor (added last). If your processor performs some kind of rendering, it should usually also include an optional `ImageInport`, where previously rendered parts of an image can be passed through, so that the processor's outport produces a composited image of the current and previous renderings.
+
+## process()
+The `process()` function is where the functionality of your processor is defined. Every time the processor is invalidated (either by changing inputs, events or property changes), the `process()` function is executed.
+Inside the `process()` you can access your inports' data by using `yourinport.getData()` and similarly you can write data to the outports using `youroutport.setData(...)`. What happens with your data in between is fully up to you.
+You can also use all your defined properties here. This let's you access all your algorithm's parameters directly from the GUI with automatic updates upon change.
+
+### Frequently used operations
+As mentioned above, most processors that output an image, should also have an optional image inport for compositing of different renders. Usually by default you want to render on top of this incoming image, which basically holds an OpenGL framebuffer. That means you usually want to copy the inport data to the outport and activate it as target framebuffer for your rendering. `utilgl::activateTargetAndClearOrCopySource(outport, image_inport)` does exactly this for you. All your rendering will be automatically composited according to the framebuffer depth layer. Note that it is good practice to deactivate the target at the end of your `process()` by using `utilgl::deactivateCurrentTarget()`.
```

Comment:
It would be nice to know where `utilgl::activateTargetAndClearOrCopySource(outport, image_inport)` is defined, i.e. name the needed include (`#include <modules/opengl/texture/textureutils.h>`).